README.md

```diff
@@ -40,364 +43,346 @@ some kind of discriminant:
+overwrite other members. While this information is invaluable for debugging, diagnostics, formatting, and serialization,
+it is far less frequently used compared to `enum` on the whole.
+
+To avoid this inconsistency, we propose instead propose introducing reverse mapping by way of the `@@iterator` built-in
+symbol:
 
 ```js
-enum Numbers { zero, one, two, three, }
+enum E {
+  A = 0,
+  B = "A",
+}
 
-typeof Numbers.zero === "number"
-Numbers.zero === 0
-Enum.getName(Numbers, 0) === "zero"
-Enum.parse(Numbers, "zero") === 0
+for (const [key, value] of E) {
+  console.log(`${key}: ${value}`);
+}
 
-// ... strings, ...
-enum HttpMethods of String { GET, PUT, POST, DELETE }
+// prints:
+//  A: 0
+//  B: A
+```
 
-typeof HttpMethods.GET === "string"
-HttpMethods.GET === "GET"
+If adopted, TypeScript would add support for `@@iterator` while eventually deprecating existing reverse mapping support.
 
-// ... booleans, ...
-enum Switch { on = true, off = false }
 
-typeof Switch.on === "boolean";
-Switch.on === true
+## `const enum`
 
-// ... symbols, ...
-enum AlphaBeta of Symbol { alpha, beta }
+TypeScript supports the concept of a `const enum` declaration, which is similar to a normal `enum` declaration except
+that enum values are inlined into their use sites. As this capability requires whole program knowledge, it is not a
+feature we intend to support in ECMAScript enums.
 
-typeof AlphaBeta.alpha === "symbol";
-AlphaBeta.alpha.toString() === "Symbol(AlphaBeta.alpha)";
 
-// ... or a mix.
-enum Mixed {
-    number = 0,
-    string = "",
-    boolean = false,
-    symbol = Symbol()
-}
+## `Symbol` values
+
+TypeScript does not currently support `Symbol` values for enums, but would add support if this feature were to be
+adopted.
+
+
+## `BigInt` values
+
+TypeScript does not currently support `BigInt` values for enums, but would add support if this feature were to be
+adopted.
+
+
+## `export default`
+
+TypeScript does not currently support `export default` on an enum, but would add support if this feature were to be
+adopted.
+
 
-// Enums can be exported:
-export enum Zoo { lion, tiger, bear };
-export default enum { up, down, left, right };
+# Future Directions
 
-// You can test for name membership using `Enum.has()`
-Enum.has(Numbers, "one") === true
-Enum.has(Numbers, "five") === false
+While this proposal is intended to be rather limited in scope, there are several potential areas for future advancement
+in the form of follow-on proposals:
 
-// You can test for value membership using `Enum.hasValue()`:
-Enum.hasValue(Numbers, 0) === true
-Enum.hasValue(Numbers, 9) === false
+- [Algebraic Data Types](#algebraic-data-types-adts)
+- [Decorators](#decorators)
+- [Auto-Initializers](#auto-initializers-1)
 
-// You can convert enums between names and values using 
-// `Enum.parse` and `Enum.format`, respectively.
-enum AToB {
-    a = "b",
-    b = "a",
+## Algebraic Data Types (ADTs)
```

Comment:
Can you link the relevant discussions/issues?

Reply:
Regarding ADTs or something else?